R/convenience-functions.R

```diff
@@ -235,21 +235,13 @@ log_shift <- function(x, offset = 0, base = exp(1)) {
 #'   example_quantile,
 #'   c("location", "target_end_date", "target_type", "horizon", "model")
 #' )
-
 set_forecast_unit <- function(data, forecast_unit) {
-
-  datacols <- colnames(data)
-  missing <- forecast_unit[!(forecast_unit %in% datacols)]
-
-  if (length(missing) > 0) {
-    warning(
-      "Column(s) '",
-      missing,
-      "' are not columns of the data and will be ignored."
-    )
-    forecast_unit <- intersect(forecast_unit, datacols)
+  data <- ensure_data.table(data)
```

Comment:
why can't you use `as.data.table` here? Have you checked other functions to see if they have similar issues?

Reply:
I think this could use as.data.table as well.. I thought I'd use the new helper function in case the object you'd use would have any important attributes. But maybe we don't really ever need that.